b2g/components/RemoteControlService.jsm

```diff
@@ -114,11 +148,67 @@ this.RemoteControlService = {
     this._port = -1;
     this._serverSocket.close();
     this._serverSocket = null;
+    this._certFingerprint = null;
     this._serverStatus = SERVER_STATUS.STOPPED;
 
     return true;
   },
 
+  // Generate PIN code for pairing, format is 4 digits
+  // 0000 and random generated number to a new string
+  // Then get last 4 characters as new PIN code
+  generatePIN: function() {
+    let padding = new Array(PINCODE_LENGTH + 1).join('0');
+    this._pin = (padding + Math.floor(Math.random() * Math.pow(10, PINCODE_LENGTH))).slice(-1 * PINCODE_LENGTH);
```

Comment:
Math.random is not cryptographically secure. Use web crypto for this https://developer.mozilla.org/en-US/docs/Web/API/RandomSource/getRandomValues